datafusion/sqllogictest/test_files/arrow_files.slt

```diff
@@ -43,6 +43,14 @@ SELECT * FROM arrow_simple
 3 baz false
 4 NULL true
 
+query ITB
```

Comment:
I recommend creating a new `.slt` file explicitly for dynamic file catalog tests rather than extending the existing tests. This would:
1. Help show that the behavior isn't changed except when dynamic catalog is enabled
2. make it easier to evaluate how much dynamic file catalog testing there was

Also, is there sufficient negative testing? Specifically that the dynamic catalog provider isn't enabled by default (which would be a security hole?)

Reply:
I create `dynamic_file.slt` to collect all the tests for URL table. However, I also keep the URL table for each file-related slt to be a negative test to ensure this feature is disabled by default.